pytorch_lightning/loops/base.py

```diff
@@ -108,8 +201,13 @@ def run(self, *args: Any, **kwargs: Any) -> Optional[Any]:
         output = self.on_run_end()
         return output
 
-    def restore(self) -> None:
-        """Restore the internal state of the loop the beginning of run if restarting is ``True``."""
+    def restore(self) -> Optional[bool]:
+        """
+        Restore the internal state of the loop the beginning of run if restarting is ``True``.
+
+        Returns:
+            is_restore_finished: Whether the restoration actually finished.
```

Comment:
in which case will a restoration not finish. 
is this this about a success/fail return signal?

Reply:
This is just a mechanism to enable a loop to delay the restarting variable to be set to False.

In `TrainingBatchLoop`, this property is required to know if we should `fast-forward` the dataloaders, therefore by return False, I indicate to the loop I will manually switch off the restarting boolean later on in the loop.

Example:
https://github.com/PyTorchLightning/pytorch-lightning/pull/8337/files#diff-eba04421e2e60c9d7b55f56ca57c9a319a4e3de38e3eb96c931324ecacbb73e0R175